src/runtime/builtin_modules/builtins.cpp

```diff
@@ -145,30 +145,77 @@ extern "C" Box* any(Box* container) {
     return boxBool(false);
 }
 
-extern "C" Box* min(Box* arg0, BoxedTuple* args) {
+Box* min_max(Box* arg0, BoxedTuple* args, BoxedDict* kwargs, int opid) {
     assert(args->cls == tuple_cls);
+    assert(kwargs->cls == dict_cls);
 
-    Box* minElement;
+    bool pass_kwargs = (kwargs && kwargs->d.size());
+    Box* key_func = nullptr;
+    Box* extremElement;
     Box* container;
+    Box* extremVal;
+
+    if (pass_kwargs) {
+        static BoxedString* key_str = static_cast<BoxedString*>(PyString_InternFromString("key"));
+        auto it = kwargs->d.find(key_str);
+        if (it != kwargs->d.end()) {
+            key_func = it->second;
+        } else {
+            if (opid == Py_GT)
+                raiseExcHelper(TypeError, "min() got an unexpected keyword argument");
+            else
+                raiseExcHelper(TypeError, "max() got an unexpected keyword argument");
+        }
```

Comment:
We need a check here that the size of kwargs was 1, so that we can reject things like `min([1], key=lambda x:x, extra_arg=1)`.